experimental/dds/tree2/src/feature-libraries/default-field-kinds/optionalField.ts

```diff
@@ -156,106 +278,162 @@ export const optionalChangeRebaser: FieldChangeRebaser<OptionalChangeset> = {
-							NodeExistenceState.Dead,
-						),
-						deletedBy: {
-							revision: overIntention,
-							localId: over.fieldChange.id,
-						},
-					};
-				} else if (over.fieldChange.newContent !== undefined) {
+
+				if (over.fieldChange.newContent !== undefined) {
 					const overContent = over.fieldChange.newContent;
-					const rebasingOverRollback =
-						overIntention === change.deletedBy.revision &&
-						over.fieldChange.id === change.deletedBy.localId;
-					const rebasingOverUndo =
-						"revert" in overContent &&
-						overContent.changeId.revision === change.deletedBy.revision &&
-						overContent.changeId.localId === change.deletedBy.localId;
-					if (rebasingOverRollback || rebasingOverUndo) {
-						// Over is reviving the node that change.childChange is referring to.
-						// Rebase change.childChange and remove deletedBy
-						// because we revived the node that childChange refers to
-						return {
-							childChange: rebaseChild(
-								change.childChange,
-								overContent.changes,
-								NodeExistenceState.Alive,
-							),
-						};
+					restoredRollbackChangeId = {
+						revision: overIntention,
+						localId: over.fieldChange.id,
+					};
+					if ("revert" in overContent) {
+						restoredUndoChangeId = overContent.changeId;
 					}
 				}
 			}
-		}
-
-		{
-			const rebasedChange = { ...change };
 
-			let overChildChange: NodeChangeset | undefined;
-			if (change.deletedBy === undefined && over.deletedBy === undefined) {
-				overChildChange = over.childChange;
+			for (const [id, childChange] of change.childChanges) {
+				if (id === "self") {
+					const overChildChange = overChildChanges.get(id);
+					if (over.fieldChange !== undefined) {
+						// `childChange` refers to the node existing in this field before rebasing, but
+						// that node was removed by `over`.
+						const rebasedChild = rebaseChild(
+							childChange,
+							overChildChange,
+							NodeExistenceState.Dead,
+						);
+						if (rebasedChild !== undefined) {
+							perChildChanges.set(
+								{
+									// TODO: Document this choice. This isn't really the revision that deleted the node, but
+									// the one that puts it back such that if we later ressurect it, the child changes will
+									// apply to it... this matches what the previous code/format did, but it's not well-documented
+									// why it's the right choice.
+									// See the "can rebase a node replacement and a dependent edit to the new node" test case.
+									// This might be making assumptions on sandwich rebasing a la rollback tags (which could be
+									// an obstacle for postbase)
+									revision: getIntention(
+										over.fieldChange?.revision ?? overTagged.revision,
+										revisionMetadata,
+									),
+									localId: over.fieldChange.id,
+								},
+								rebasedChild,
+							);
+						}
+					} else {
+						// `over` didn't remove the node (its fieldChange is undefined), and so `overChildChange`
```

Comment:
`overChildChange` refers to the same node as `childChange` even if `over` did remove the node, right?

Reply:
Yeah, my understanding of the point of this comment (@Abe27342 can correct me if I'm wrong) is that `over` didn't remove the node. I'm not sure the second part needs to be said because of the block we're in.